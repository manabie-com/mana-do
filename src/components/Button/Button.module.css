.root {
  line-height: 1.5;
  font-weight: 400;
  text-align: center;
  box-shadow: 0 2px 0 rgba(0,0,0,.015);
  cursor: pointer;
  transition: all .3s linear;
  height: 36px;
  padding: 4px 18px;
  font-size: 14px;
  border-radius: 2px;
  color: rgba(0,0,0,.85);
  border: 1px solid #d9d9d9;
  background: #fff;
}

.root:hover {
  color: #40a9ff;
  border-color: #40a9ff;
  background: #fff;
}