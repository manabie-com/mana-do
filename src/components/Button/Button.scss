@import "../../assets/scss/variables";

.btn {
  background-color: $primiry-color;
  font-weight: 700;
  height: 46px;
  width: 100%;
  max-width: 150px;
  border: none;
  font-size: 16px;
  color: white;
  border-radius: 6px;
  cursor: pointer;
  &--red {
    background-color: white;
    margin-top: 16px;
    color: $error-color;
    border: solid 1px $error-color;
    transition: all 500ms;
    &:hover {
      background-color: $error-color;
      color: white;
      border-color: $error-color;
      transition: all 500ms;
    }
  }
  &--white {
    background-color: white;
    color: $primiry-color;
    border: solid 1px $primiry-color;
  }
}
