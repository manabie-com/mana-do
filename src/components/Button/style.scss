.custom-button {
  button {
    background-color: gray;
    cursor: pointer;
    font-weight: bold;
    outline: none;
    border: 1px solid transparent;
    margin: 0px auto;
    display: block;
    border-radius: 4px;
    min-width: 80px;
    color: white;
    transition: all 0.5s;
    &:hover {
      background-color: white;
      opacity: 0.85;
      border: 1px solid gray;
      color: gray;
    }
  }
  //size
  .btn--size {
    &--large {
      padding: 12px 8px;
    }
    &--medium {
      padding: 8px 4px;
    }
    &--small {
      padding: 8px;
      min-width: 0px;
    }
  }
  //color
  .btn--color {
    &--primary {
      background-color: #007bff;
      &:hover {
        color: #007bff;
        border: 1px solid #007bff;
      }
    }
    &--success {
      background-color: #28a745;
      &:hover {
        color: #28a745;
        border: 1px solid #28a745;
      }
    }
    &--danger {
      background-color: #dc3545;
      &:hover {
        color: #dc3545;
        border: 1px solid #dc3545;
      }
    }
  }
}
.btn--active {
  .btn--color {
    &--primary {
      color: #007bff;
      border: 1px solid #007bff;
      background: transparent;
    }
    &--success {
      color: #28a745;
      border: 1px solid #28a745;
      background: transparent;
    }
    &--default{
      color: gray;
      border: 1px solid gray;
      background: transparent;
    }
  }
}
