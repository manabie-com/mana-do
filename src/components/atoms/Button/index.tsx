// Libraries
import React from 'react';
import {MAP_STYLE_COLOR} from "./constants";

interface TButtonProps extends React.HTMLAttributes<any> {
    className?: string
    color?: "primary" | "success" | "errors" | "warning"
}

export const Button: React.FC<TButtonProps> = (props) => {
    // props
    const {children, color="primary"} = props

    return (
        <button {...props} className={`${props.className || ''} btn ${MAP_STYLE_COLOR[color]}`}>
            {children}
        </button>
    );
};
