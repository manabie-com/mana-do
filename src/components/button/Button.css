
.btn {background-image: linear-gradient(to right, #DA22FF 0%, #9733EE  51%, #DA22FF  100%)}
.btn {
  margin: 0px;
  text-align: center;
  text-transform: uppercase;
  transition: 0.5s;
  background-size: 200% auto;
  color: white;
  box-shadow: 0 0 20px #eee;
 }

 .btn:hover {
   background-position: right center; /* change the direction of the change here */
   color: #fff;
   text-decoration: none;
 }

 .btn-danger {
   background:#ff4d4f ;
 }