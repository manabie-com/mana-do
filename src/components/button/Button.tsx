import React, {
  ButtonHTMLAttributes,
  FC,
  memo,
  MouseEventHandler,
} from "react";
import "./styles.scss";

export interface IButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
  disabled?: boolean;
  onClick: MouseEventHandler<HTMLButtonElement> | undefined;
}

const Button: FC<IButtonProps> = ({
  className,
  children,
  disabled,
  onClick,
  ...props
}) => {
  return (
    <button
      className={`btn-todo ${className ? className : ""}`}
      onClick={onClick}
      disabled={disabled}
      data-testid="button"
      {...props}
    >
      {children}
    </button>
  );
};

export default memo(Button);
