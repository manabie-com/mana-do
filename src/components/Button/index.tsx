import React from 'react'
import './Button.css'

const Button = (props: any) => {
    const { className, title, onClick, disabled, active } = props

    return (
        <button className={`${className} ${active && "btn__active"} `} onClick={onClick} disabled={disabled} type="button">
            {title}
        </button>
    )
}

export default Button