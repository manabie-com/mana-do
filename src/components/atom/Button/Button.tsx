import React from 'react';
import './button.scss'

export enum ButtonType {
    SUBMIT = "submit",
    BUTTON = "button",
    RESET = "reset",
}
type Button = ButtonType | undefined

interface IButton {
    label: string,
    className: string,
    type?: Button,
    style?: any,
    onClick?: (e: any) => void
    onBlur?: (e: any) => void
}
const Button = ({ onBlur, onClick, label, className, type, style }: IButton) => {
    const handleOnClick = (e: any) => {
        if (onClick) {
            onClick(e)
        }
    }
    const handleOnBlur = (e: any) => {
        if (onBlur) {
            onBlur(e)
        }
    }

    return (
        <button
            data-testid={className}
            style={style}
            className={`button ${className}`}
            type={type}
            onClick={handleOnClick}
            onBlur={handleOnBlur}
        >
            {label}
        </button >
    )
}
export default Button;
