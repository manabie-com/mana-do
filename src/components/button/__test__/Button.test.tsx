import { render, RenderResult } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { getElementByTestId } from "utils/testUtils";
import Button, { IButtonProps } from "../Button";

const onClickAction = jest.fn();

const renderComponent = ({ ...props }: IButtonProps): RenderResult => {
  return render(<Button {...props} />);
};

describe("<Button /> rendering", () => {
  it("should match snapshot", () => {
    const wrapper = renderComponent({
      onClick: onClickAction,
    });
    expect(wrapper.container).toMatchSnapshot();
  });

  it("should render correcly", () => {
    renderComponent({
      onClick: onClickAction,
    });
    const element = getElementByTestId("button");
    expect(element).toBeInTheDocument();
  });

  it("should render a button with the default class", () => {
    renderComponent({
      onClick: onClickAction,
    });
    const element = getElementByTestId("button");
    expect(element).toHaveClass("btn-todo");
  });
});

describe("<Button /> interacting", () => {
  it("should render a button clickable", () => {
    renderComponent({
      onClick: onClickAction,
    });
    const element = getElementByTestId("button");
    expect(element).not.toBeDisabled();
  });

  it("should handle onClick when clicked", () => {
    renderComponent({
      onClick: onClickAction,
    });
    const element = getElementByTestId("button");
    userEvent.click(element);
    expect(onClickAction).toHaveBeenCalled();
  });
});
