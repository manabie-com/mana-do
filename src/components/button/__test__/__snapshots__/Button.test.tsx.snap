// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> rendering should match snapshot 1`] = `
<div>
  <button
    class="btn-todo "
    data-testid="button"
  />
</div>
`;
